packages/multichain-account-service/src/MultichainAccountWallet.ts

```diff
@@ -326,4 +326,101 @@ export class MultichainAccountWallet<
       await group.align();
     }
   }
+
+  /**
+   * Discover and create accounts for all providers.
+   *
+   * NOTE: This method should only be called on a newly created wallet.
+   *
+   * @param opts - The options for the discovery and creation of accounts.
+   * @param opts.skippedProviders - The providers to skip.
+   * @returns The accounts for each provider.
+   */
+  async discoverAndCreateAccounts({
+    skippedProviders = [],
+  }: {
+    skippedProviders: AccountProviderType[];
+  }): Promise<{
+    [AccountProviderType]: Bip44Account<KeyringAccount>[];
+  }> {
+    const providers = this.#providers.filter(
+      (provider) => !skippedProviders.includes(provider.providerType),
+    );
+
+    const results: Record<AccountProviderType, Bip44Account<KeyringAccount>[]> =
+      {};
+    const nextIndex = new Map<Bip44Provider, number>(
+      providers.map((p) => [p, 0]),
+    );
+    const stopped = new Set<Bip44Provider>();
+    const inFlight = new Map<Bip44Provider, Promise<void>>();
```

Comment:
WDYT about using a single `Map` for this with something like:
```ts
const contexts = new Map<Bip44Provider, {
  stopped: boolean;
  running: Promise<void>;
  groupIndex: number;
}>();
```

Reply:
We could even add the `accounts` in there too, so we can extract `results` from it afterward.

Reply:
Also, using an object might remove every `?? 0` we have 🤔